satpy/writers/__init__.py

```diff
@@ -480,7 +480,8 @@ def to_image(dataset):
         Instance of :class:`~trollimage.xrimage.XRImage`.
 
     """
-    dataset = dataset.squeeze()
+    if not dataset.coords.get("bands") == ["P"]:
```

Comment:
```suggestion
    if dataset.coords.get("bands") != ["P"]:
```

Or just remove it?

Reply:
bit early for reviews, I don't know what I'm doing yet :)

It looks like I shouldn't even have a mode `P` at this point because then I won't be able to use `XRImage.palettize` later, as that only accepts L or LA